web/docs/tql2/operators/http.md

```diff
@@ -0,0 +1,218 @@
+
+Field to insert metadata into when using the parsing pipeline.
+
+The metadata has the following schema:
+
+| Field                | Type     | Description                           |
+| :------------------- | :------- | :------------------------------------ |
+| `code`               | `uint64` | The HTTP status code of the response. |
+| `headers`            | `record` | The response headers.                 |
+
+### `paginate = string (optional)`
+
+An expression to evaluate against the result of the request (optionally parsed
+by the given pipeline). If the expression evaluation is successful and non-null, the
+resulting string is used as the URL for a new GET request with the same headers.
+
+### `paginate_delay = duration (optional)`
+
+The duration to wait between consecutive paginatation requests.
+
+Defaults to `0s`.
+
+### `parallel = int (optional)`
+
+Maximum amount of requests that can be in progress at any time.
+
+Defaults to `1`.
+
+### `tls = bool (optional)`
+
+Enables TLS.
+
+### `certfile = string (optional)`
+
+Path to the client certificate.
+
+### `keyfile = string (optional)`
+
+Path to the key for the client certificate.
+
+### `password = string (optional)`
+
+Password for keyfile.
+
+### `connection_timeout = duration (optional)`
+
+Timeout for the connection.
+
+Defaults to `5s`.
+
+### `max_retry_count = int (optional)`
+
+The maximum times to retry a failed request. Every request has its own retry
+count.
+
+Defaults to `0`.
+
+### `retry_delay = duration (optional)`
+
+Duration to wait between each retry.
+
+Defaults to `1s`.
+
+### `{ … }`
+
+A pipeline that receives the response body as bytes, allowing parsing per
+request. This is especially useful in scenarios where the response body can be
+parsed into multiple events.
+
+## Examples
+
+### Make a GET request
+
+Here we make a request to [urlscan.io](https://urlscan.io/docs/api#search) to search for scans for `tenzir.com` and get the first result.
+
+```tql
+from {}
+http "https://urlscan.io/api/v1/search?q=tenzir.com" {read_json}
+unroll results
+head 1
+```
+
+```tql
+{
+  results: {
+    submitter: { ... },
+    task: { ... },
+    stats: { ... },
+    page: { ... },
+    _id: "0196edb1-521e-761f-9d62-1ca4cfad5b30",
+    _score: null,
+    sort: [ "1747744570133", "\"0196edb1-521e-761f-9d62-1ca4cfad5b30\"" ],
+    result: "https://urlscan.io/api/v1/result/0196edb1-521e-761f-9d62-1ca4cfad5b30/",
+    screenshot: "https://urlscan.io/screenshots/0196edb1-521e-761f-9d62-1ca4cfad5b30.png",
+  },
+  total: 9,
+  took: 296,
+  has_more: false,
+}
+```
+
+### Paginate an API
+
+In the above example, we can utilize the `sort` and `has_more` fields to get
+more pages from the API. 
+
+```tql
+let $URL = "https://urlscan.io/api/v1/search?q=example.com"
+from {}
+http $URL, paginate=$URL + "&search_after=" + encode_url(results.last().sort.first() + "," + results.last().sort.last().slice(begin=1, end=-1)) if has_more? {
+  read_json
+}
+head 10
```

Comment:
Can we add one more example of a paginated api with a less complicated next page identifier extraction process? This one is nice to show the full power, but it is quite difficult to tell the upstream requirements from the operators logic.

Reply:
Will be done in a follow-up.